### PR TITLE
Improve AWS-related config for Sage profile

### PR DESCRIPTION
**Pull request by BrunoGrandePhD**
Over the past few months, we've discovered several config optimizations for running our workflows on AWS Batch. Because the resource allocation tweaks haven't been tested thoroughly, I'm disabling them here with the intent of enabling them later once they're tested.

### CODE REVIEW

**Comment by BrunoGrandePhD** [comment]
Requesting a review from @nf-core/configs-team. Thanks!

**Review by jfy133 — CHANGES_REQUESTED** [review]
1. If you don't need it, you should remove the resource allocation tweaks sectin - this should really be done on a per-pipeline basis
2. Related to 1 - you shouldn't need the `check_max` function as again should already be in each pipeline's `nextflow.config`

**Comment by BrunoGrandePhD** [comment]
Thanks for the review, @jfy133! 

> 1. If you don't need it, you should remove the resource allocation tweaks sectin - this should really be done on a per-pipeline basis

I opted to uncomment that section and simplify it instead by removing the Sarek-specific section. I agree that optimizations should be done at the pipeline level (you can expect future PRs on this), but we also have some general tweaks that should apply to all pipelines. For example, we want to universally increase the time limits on jobs because it sucks when a long-running job gets killed prematurely. That's costly for a non-profit like us. 

> 2. Related to 1 - you shouldn't need the `check_max` function as again should already be in each pipeline's `nextflow.config`

As I mention in the comment above the `check_max` function, it isn't present in Sarek 2.7. Instead, they have a `check_resource` [function](https://github.com/nf-core/sarek/blob/e8f56e5bbb6f8a34793a6b5a2945981eb43090aa/nextflow.config#L255-L269). I also [tried](https://github.com/nf-core/configs/commit/6f0d9e6c43b8ee0373011f32d4dbf3c17811cd78) including this function in the Sarek-specific config, but it didn't work, presumably because of the evaluation order.

**Review by thomasyu888 — APPROVED** [review]

File: conf/sage.config [hunk 1]
```diff
@@ -1,19 +1,52 @@
...
+params {
+  igenomes_base = 's3://sage-igenomes/igenomes'
+  max_memory    = '128.GB'
+  max_cpus      = 16
+  max_time      = '240.h'
+}
+
+// Enable retries globally for certain exit codes
+process {
+  errorStrategy = { task.exitStatus in [143,137,104,134,139,247] ? 'retry' : 'finish' }
```
Comment (BrunoGrandePhD) [inline conf/sage.config:18]: I wonder if there's a way to use this same strategy to adjust whether to increase the time allocation or memory based on the exit code. For example, on AWS Batch, exit code 143 means that the job reached its time limit.

**Comment by jfy133** [comment]
> Thanks for the review, @jfy133!
> 
> > 1. If you don't need it, you should remove the resource allocation tweaks sectin - this should really be done on a per-pipeline basis
> 
> I opted to uncomment that section and simplify it instead by removing the Sarek-specific section. I agree that optimizations should be done at the pipeline level (you can expect future PRs on this), but we also have some general tweaks that should apply to all pipelines. For example, we want to universally increase the time limits on jobs because it sucks when a long-running job gets killed prematurely. That's costly for a non-profit like us.

OK fair enough - then lets leave it until you start getting the pipeline specific configs ready

> > 2. Related to 1 - you shouldn't need the `check_max` function as again should already be in each pipeline's `nextflow.config`
> 
> As I mention in the comment above the `check_max` function, it isn't present in Sarek 2.7. Instead, they have a `check_resource` [function](https://github.com/nf-core/sarek/blob/e8f56e5bbb6f8a34793a6b5a2945981eb43090aa/nextflow.config#L255-L269). I also [tried](https://github.com/nf-core/configs/commit/6f0d9e6c43b8ee0373011f32d4dbf3c17811cd78) including this function in the Sarek-specific config, but it didn't work, presumably because of the evaluation order.

Sarek :shakesfist: OK fair enough

**Review by jfy133 — APPROVED** [review]

File: conf/sage.config (hunk 1 above)
Comment (jfy133) [reply conf/sage.config:18]: That's a very good question actuallly... might be worth asking on slack, it wouldn't surprise me if someone hasn't already had that issue.

**Comment by jfy133** [comment]
You can ignore the 'repository root' test failure btw

**Pull request merged**